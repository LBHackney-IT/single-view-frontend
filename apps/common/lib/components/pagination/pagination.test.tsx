import React from "react";

import { render, testA11y } from "@hackney/mtfh-test-utils";

import {
  Pagination,
  PaginationButton,
  PaginationControls,
  PaginationSummary,
} from "./pagination";

test("it renders correctly", async () => {
  const { container } = render(
    <Pagination>
      <PaginationSummary>Showing 101â€”150 of 246 results</PaginationSummary>
      <PaginationControls>
        <PaginationButton variant="previous" href="/2">
          Previous
        </PaginationButton>
        <PaginationButton href="/1">1</PaginationButton>
        <PaginationButton href="/2">2</PaginationButton>
        <PaginationButton href="/3" active>
          3
        </PaginationButton>
        <PaginationButton href="/4">4</PaginationButton>
        <PaginationButton href="/5">5</PaginationButton>
        <PaginationButton variant="next" href="/4">
          Next
        </PaginationButton>
      </PaginationControls>
    </Pagination>,
  );
  expect(container).toMatchSnapshot();
  await testA11y(container);
});

test("it renders without a summary", async () => {
  const { container } = render(
    <Pagination>
      <PaginationControls>
        <PaginationButton variant="previous" href="/2">
          Previous
        </PaginationButton>
        <PaginationButton href="/1">1</PaginationButton>
        <PaginationButton href="/2">2</PaginationButton>
        <PaginationButton variant="next" href="/4">
          Next
        </PaginationButton>
      </PaginationControls>
    </Pagination>,
  );
  expect(container).toMatchSnapshot();
  await testA11y(container);
});

test("it renders control as buttons", async () => {
  const { container } = render(
    <Pagination>
      <PaginationControls>
        <PaginationButton variant="previous" href="/2">
          Previous
        </PaginationButton>
        <PaginationButton as="button">1</PaginationButton>
        <PaginationButton variant="next" href="/4">
          Next
        </PaginationButton>
      </PaginationControls>
    </Pagination>,
  );
  expect(container).toMatchSnapshot();
  await testA11y(container);
});
