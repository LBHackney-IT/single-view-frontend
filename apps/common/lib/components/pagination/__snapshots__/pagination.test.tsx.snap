// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders control as buttons 1`] = `
<div>
  <nav
    class="lbh-pagination"
  >
    <ul
      class="lbh-pagination__list"
    >
      <li
        class="lbh-pagination__item"
      >
        <a
          class="lbh-pagination__link"
          href="/2"
          rel="previous"
        >
          <span
            aria-hidden="true"
            role="presentation"
          >
            «
             
          </span>
          Previous
        </a>
      </li>
      <li
        class="lbh-pagination__item"
      >
        <button
          class="lbh-pagination__link"
        >
          1
        </button>
      </li>
      <li
        class="lbh-pagination__item"
      >
        <a
          class="lbh-pagination__link lbh-pagination__link--next"
          href="/4"
          rel="next"
        >
          Next
          <span
            aria-hidden="true"
            role="presentation"
          >
             
            »
          </span>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`it renders correctly 1`] = `
<div>
  <nav
    class="lbh-pagination"
  >
    <div
      class="lbh-pagination__summary"
    >
      Showing 101—150 of 246 results
    </div>
    <ul
      class="lbh-pagination__list"
    >
      <li
        class="lbh-pagination__item"
      >
        <a
          class="lbh-pagination__link"
          href="/2"
          rel="previous"
        >
          <span
            aria-hidden="true"
            role="presentation"
          >
            «
             
          </span>
          Previous
        </a>
      </li>
      <li
        class="lbh-pagination__item"
      >
        <a
          class="lbh-pagination__link"
          href="/1"
        >
          1
        </a>
      </li>
      <li
        class="lbh-pagination__item"
      >
        <a
          class="lbh-pagination__link"
          href="/2"
        >
          2
        </a>
      </li>
      <li
        class="lbh-pagination__item"
      >
        <a
          class="lbh-pagination__link lbh-pagination__link--current"
          href="/3"
        >
          3
        </a>
      </li>
      <li
        class="lbh-pagination__item"
      >
        <a
          class="lbh-pagination__link"
          href="/4"
        >
          4
        </a>
      </li>
      <li
        class="lbh-pagination__item"
      >
        <a
          class="lbh-pagination__link"
          href="/5"
        >
          5
        </a>
      </li>
      <li
        class="lbh-pagination__item"
      >
        <a
          class="lbh-pagination__link lbh-pagination__link--next"
          href="/4"
          rel="next"
        >
          Next
          <span
            aria-hidden="true"
            role="presentation"
          >
             
            »
          </span>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`it renders without a summary 1`] = `
<div>
  <nav
    class="lbh-pagination"
  >
    <ul
      class="lbh-pagination__list"
    >
      <li
        class="lbh-pagination__item"
      >
        <a
          class="lbh-pagination__link"
          href="/2"
          rel="previous"
        >
          <span
            aria-hidden="true"
            role="presentation"
          >
            «
             
          </span>
          Previous
        </a>
      </li>
      <li
        class="lbh-pagination__item"
      >
        <a
          class="lbh-pagination__link"
          href="/1"
        >
          1
        </a>
      </li>
      <li
        class="lbh-pagination__item"
      >
        <a
          class="lbh-pagination__link"
          href="/2"
        >
          2
        </a>
      </li>
      <li
        class="lbh-pagination__item"
      >
        <a
          class="lbh-pagination__link lbh-pagination__link--next"
          href="/4"
          rel="next"
        >
          Next
          <span
            aria-hidden="true"
            role="presentation"
          >
             
            »
          </span>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
